CRM/Financial/Form/BatchTransaction.php

```diff
@@ -52,7 +52,17 @@ public function preProcess() {
     $this->assign('entityID', self::$_entityID);
     if (isset(self::$_entityID)) {
       $this->_batchStatusId = CRM_Core_DAO::getFieldValue('CRM_Batch_BAO_Batch', self::$_entityID, 'status_id');
+      $this->batchStatuses = CRM_Core_OptionGroup::values('batch_status', TRUE);
```

Comment:
1. This will Fetch option values with Value as option value Label, we need to match with machine name. 
2. I would recommend to use CRM_Core_Pseudoconstant::get() method to retrieve Batch statuses.
3. From UI/API its possible to change the Option group name, so flip() won't be a good idea i guess.